### PR TITLE
Notification bar for Buyer.Web

### PR DESCRIPTION
**Pull request by PiotrZaranek**

### CODE REVIEW

File: fe/projects/AspNetCore/src/areas/DownloadCenter/pages/DownloadCenterCategoryPage/DownloadCenterCategoryPage.js [hunk 1]
```diff
@@ -18,6 +19,9 @@ const DownloadCenterCategoryPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme(props.locale)}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/DownloadCenter/pages/DownloadCenterCategoryPage/DownloadCenterCategoryPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/DownloadCenter/pages/DownloadCenterPage/DownloadCenterPage.js [hunk 2]
```diff
@@ -17,6 +18,9 @@ const DownloadCenterPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme()}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/DownloadCenter/pages/DownloadCenterPage/DownloadCenterPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Home/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -20,6 +21,9 @@ function HomePage(props) {
     <ThemeProvider theme={GlobalHelper.initMuiTheme()}>
       <ToastContainer />
       <Store>
+        {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Home/pages/HomePage/HomePage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/News/pages/NewsItemPage/NewsItemPage.js [hunk 4]
```diff
@@ -18,6 +19,9 @@ const NewsItemPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme()}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/News/pages/NewsItemPage/NewsItemPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/News/pages/NewsPage/NewsPage.js [hunk 5]
```diff
@@ -17,6 +18,9 @@ const NewsPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme()}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/News/pages/NewsPage/NewsPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Orders/pages/ListOrders/ListOrdersPage.js [hunk 6]
```diff
@@ -17,6 +18,9 @@ const ListOrdersPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme(props.locale)}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Orders/pages/ListOrders/ListOrdersPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Orders/pages/NewOrder/NewOrderPage.js [hunk 7]
```diff
@@ -17,6 +18,9 @@ const NewOrderPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme(props.locale)}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Orders/pages/NewOrder/NewOrderPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Orders/pages/OrderItemPage/OrderItemPage.js [hunk 8]
```diff
@@ -17,6 +18,9 @@ const OrderItemPage = (props) => {
       <ThemeProvider theme={GlobalHelper.initMuiTheme(props.locale)}>
         <ToastContainer />
         <Store>
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Orders/pages/OrderItemPage/OrderItemPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Orders/pages/StatusOrder/StatusOrderPage.js [hunk 9]
```diff
@@ -17,6 +18,9 @@ const StatusOrderPage = (props) => {
         <ThemeProvider theme={GlobalHelper.initMuiTheme(props.locale)}>
             <ToastContainer />
             <Store>
+                {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Orders/pages/StatusOrder/StatusOrderPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Products/pages/AvailableProductsPage/AvailableProductsPage.js [hunk 10]
```diff
@@ -18,6 +19,9 @@ function AvailableProductsPage(props) {
       <ToastContainer />
       <Store>
         <div className="search-products-page">
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/pages/AvailableProductsPage/AvailableProductsPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Products/pages/CategoryPage/CategoryPage.js [hunk 11]
```diff
@@ -19,6 +20,9 @@ function CategoryPage(props) {
       <ToastContainer />
       <Store>
         <div className="category-page">
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/pages/CategoryPage/CategoryPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Products/pages/OutletPage/OutletPage.js [hunk 12]
```diff
@@ -15,6 +16,9 @@ function OutletPage(props) {
       <ToastContainer />
       <Store>
         <div className="search-products-page">
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/pages/OutletPage/OutletPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Products/pages/ProductPage/ProductPage.js [hunk 13]
```diff
@@ -19,6 +20,9 @@ function ProductPage(props) {
       <ToastContainer />
       <Store>
         <div className="category-page">
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/pages/ProductPage/ProductPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/areas/Products/pages/SearchProductsPage/SearchProductsPage.js [hunk 14]
```diff
@@ -7,13 +7,17 @@ import Header from "../../../../shared/components/Header/Header";
...
 import Footer from "../../../../shared/components/Footer/Footer";
+import NotificationBar from "../../../../shared/components/NotificationBar/NotificationBar";
 
 function SearchProductsPage(props) {
   return (
     <ThemeProvider theme={GlobalHelper.initMuiTheme()}>
       <ToastContainer />
       <Store>
         <div className="search-products-page">
+          {props.notificationBar.items &&
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/pages/SearchProductsPage/SearchProductsPage.js]: **suggestion:** Please add condition for notification bar, should be:

```js
{props.notificationBar && props.notificationBar.items
```

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.scss [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    &_link {
+        color: $white;
+
+        &:hover {
+            color: $lightgrey;
+        }
+    }
+}
+
+@media only screen and (max-width: 1099px) {
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.scss]: **suggestion:** Please use bulma variables

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.scss [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+            display: block;
+        }
+
+        &_desktop {
+            display: none;
+        }
+    }
+}
+
+@media only screen and (min-width: 1100px) {
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.scss]: **suggestion:** Please use bulma variables

File: fe/projects/AspNetCore/src/stories/10-Home.stories.js [hunk 17]
```diff
@@ -3,6 +3,33 @@ import "../areas/Home/pages/HomePage/HomePage.scss";
 import HomePage from "../areas/Home/pages/HomePage/HomePage";
 import { header, mainNavigation, footer } from "./shared/Props";
 
+var notificationBar = {
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/stories/10-Home.stories.js:6]: **suggestion:**  Please add example data in english language

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+import * as Icon from "@mui/icons-material";
+import { Splide, SplideSlide } from "@splidejs/react-splide";
+
+const NotificationBar = (props) => {
+
+    const getIconTag = (iconName) => {
+        const IconTag = Icon[iconName];
+
+        if (IconTag === undefined || IconTag === null) {
+            IconTag = Icon["PushPin"];
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js]: **suggestion:** Please add PushPin to constants

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        return (
+            <span>
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js]: **suggestion:** You can return without span element

File: be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar.json [hunk 20]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (SebaKas21) [inline be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar.json:1]: **suggestion (non-blocking):** Add limitation for elements in items. For example: 3

File: be/src/Project/Web/Buyer/Buyer.Web/Shared/ViewModels/Base/BasePageViewModel.cs [hunk 21]
```diff
@@ -1,14 +1,14 @@
...
 
 namespace Buyer.Web.Shared.ViewModels.Base
 {
     public class BasePageViewModel
     {
         public string Locale { get; set; }
-        public HeaderViewModel Header { get; set; }
-        public MenuTilesViewModel MenuTiles { get; set; }
+        public NotificationBarViewModel NotificationBar { get; set; }
+        public BuyerHeaderViewModel Header { get; set; }
```
Comment (SebaKas21) [inline be/src/Project/Web/Buyer/Buyer.Web/Shared/ViewModels/Base/BasePageViewModel.cs:12]: **suggestion:** Missing element for BasePageVIewModel

```cs
public MainNavigationViewModel MainNavigation { get; set; }
```

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+                        <div className="notification-bar_item is-flex">
+                            {getIconTag(item.icon)}
+                            <a href={item.link.url} target={item.link.target} className="notification-bar_link">
+                                {item.link.text}
+                            </a>
+                        </div>
+                    ))}
+                </div>
+            </div>
+            <div className="notification-bar_mobile">
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js:36]: **suggestion:** When is only one element in items. Displayed:

obraz

File: be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar-item.json [hunk 23]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (SebaKas21) [inline be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar-item.json:1]: **suggestion:** Add required attribute for icon or add default option like PushPin. When you set default option, you can ignore my suggestion with constants in notification bar component

**Review by SebaKas21 — CHANGES_REQUESTED** [review]
Hi @PiotrZaranek 

Suggestion for PR name: "Notification bar for Buyer.Web"

File: be/src/Project/Services/Content/Content.Api/src/api/home-page/content-types/home-page/schema.json [hunk 24]
```diff
@@ -55,6 +55,16 @@
         }
       },
       "component": "shared.footer"
+    },
+    "notificationBar": {
```
Comment (SebaKas21) [inline be/src/Project/Services/Content/Content.Api/src/api/home-page/content-types/home-page/schema.json:59]: **suggestion (non-blocking):** Should be notificationBarItems

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar-item.json (hunk 23 above)
Comment (PiotrZaranek) [reply be/src/Project/Services/Content/Content.Api/src/components/blocks/notification-bar-item.json:1]: This resolution doesn't resolve problem when user set wrong name of icon. Bad name of icon set in CMS causes error of the entire website.

**Comment by dawiddworak88** [comment]
Hi @PiotrZaranek, could you please resolve the conflicts and test the functionality once again? Once you're happy with all the changes I'll review it and we'll merge your changes to the develop branch

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 25]
```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as Icon from "@mui/icons-material";
+import { Splide, SplideSlide } from "@splidejs/react-splide";
+
+const NotificationBar = (props) => {
+
+    const getIconTag = (iconName) => {
+        let IconTag = Icon[iconName];
+        const defaultIconName = "PushPin";
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js]: **suggestion:** Add **PushPin** to constants

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 26]
```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as Icon from "@mui/icons-material";
+import { Splide, SplideSlide } from "@splidejs/react-splide";
+
+const NotificationBar = (props) => {
+
+    const getIconTag = (iconName) => {
+        let IconTag = Icon[iconName];
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js]: **suggestion:** Check one more time with property iconName value: null and empty string

File: fe/projects/AspNetCore/src/stories/10-Home.stories.js [hunk 27]
```diff
@@ -3,6 +3,25 @@ import "../areas/Home/pages/HomePage/HomePage.scss";
...
+  items: [
+    {
+      icon: "LocalShipping",
+      link: {
+        url: "test",
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/stories/10-Home.stories.js]: **suggestion (non-blocking):** Replace **test** with **#**

File: fe/projects/AspNetCore/src/stories/10-Home.stories.js [hunk 28]
```diff
@@ -3,6 +3,25 @@ import "../areas/Home/pages/HomePage/HomePage.scss";
...
+        text: "Up to 30 days to return",
+      }
+    },
+    {
+      icon: "LocalPostOffice",
+      link: {
+        url: "test",
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/stories/10-Home.stories.js]: **suggestion (non-blocking):** Replace **test** with **#**

**Review by SebaKas21 — CHANGES_REQUESTED** [review]
Hi @PiotrZaranek 

Let's resolve last changes and test one more time

File: fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js [hunk 29]
```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as Icon from "@mui/icons-material";
+import { Splide, SplideSlide } from "@splidejs/react-splide";
+
+const NotificationBar = (props) => {
+
+    const getIconTag = (iconName) => {
```
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/shared/components/NotificationBar/NotificationBar.js:8]: **suggestion (non-blocking):** What you think about it?:

```js
const getIconTag = (iconName) => {
    if (iconName && Icon[iconName]) {
        const IconTag = Icon[iconName];
        if (IconTag) {
            return <IconTag />;
        }
    }
    return;
};
```

**Review by SebaKas21 — APPROVED** [review]

**Review by PiotrZaranek — COMMENTED** [review]
Hi @dawiddworak88, the functionality passed the tests, you can make the final check.

**Pull request merged**